prereqScraper.py

```diff
@@ -0,0 +1,174 @@
+        title = panel.find_all(class_="courseTitleNoBlueLink")[
+            0].get_text().strip()
+        desc = panel.find_all(
+            class_="col-xs-12")[0].get_text().strip().replace("\n", "").replace("\r", "")
+        location = panel.find_all("img")[0]["alt"]
+        for child in panel.find_all(class_="col-xs-12"):
+            if child.find_all("strong"):
+                strongText = child.find_all("strong")[0].get_text()
+                if strongText == "Antirequisite(s):":
+                    for item in child.children:
+                        if hasattr(item, "children"):
+                            for req in item.children:
+                                reqTraits = [req.get_text().strip().replace(
+                                    "\n", "").replace("\r", ""), False]
+                                if not reqTraits[0]:
+                                    continue
+                                if type(req) == Tag:
+                                    if req.name == "a":
+                                        reqTraits[1] = True
+                                        antireqsLink.append(
+                                            formatLink(reqTraits[0]))
+                                antireqs.append(
+                                    {"text": reqTraits[0], "isLink": reqTraits[1]})
+                elif strongText == "Prerequisite(s):" or "Corequisite(s):" in strongText:
+                    for item in child.children:
+                        if hasattr(item, "children"):
+                            for req in item.children:
+                                reqTraits = [req.get_text().strip().replace(
+                                    "\n", "").replace("\r", ""), False]
+                                if not reqTraits[0]:
+                                    continue
+                                if "Prerequisite(s):" in reqTraits[0]:
+                                    continue
+                                if "Pre-or Corequisite(s)" in reqTraits[0]:
+                                    pcreqFlag = True
+                                elif "Corequisite(s)" in reqTraits[0]:
+                                    coreqFlag = True
+                                if type(req) == Tag:
+                                    if req.name == "a":
+                                        reqTraits[1] = True
+                                if pcreqFlag and coreqFlag:
+                                    print(title + " EXCEPTION")
+                                    raise Exception("BOTH FLAGS ACTIVE")
+                                elif pcreqFlag:
+                                    preAndCoreqs.append(
+                                        {"text": reqTraits[0], "isLink": reqTraits[1]})
+                                    if reqTraits[1]:
+                                        preAndCoreqsLink.append(
+                                            formatLink(reqTraits[0]))
```

Comment:
the links are being stored separately from the whole text itself so that we can easily relate the courses to each other. are u confused about why I didnt just grab the links from existing "isLink" part of the JSON? cause its easier to already have them this way instead of looping through the JSON afterward and finding all the links